backend/src/crd/controller/viewer/reconciler/reconciler.go

```diff
@@ -0,0 +1,274 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package reconciler describes a Reconciler for working with Viewer CRDs. The
+// reconciler takes care of the main logic for ensuring every Viewer CRD
+// corresponds to a unique deployment and backing service. The service is
+// annotated such that it is compatible with Ambassador managed routing.
```

Comment:
is it a requirement to have ambassador as dependency? Can we decouple pipeline from ambassador?

Reply:
As discussed offline, I removed explicit references to Ambassador. This now works both with and without (via direct port forwarding). I added instructions in the README for how to use the latter approach as well.